README.md

```diff
@@ -91,6 +92,37 @@ a `constructor` property which is a reference to the type representative.
 
 ## Algebras
 
+### Enum
+
+1. `E.fromEnum(E.a)` is a natural number (countability)
+2. `E.toEnum(E.fromEnum(E.a)) === E.a` (bijection)
+
+#### `fromEnum` and `toEnum` functions
+
+```hs
+fromEnum :: Enum a => a -> Natural
+toEnum :: Enum a => Natural -> a
+```
+A value which has an Enum must provide both a `fromEnum` function and
+a `toEnum` function on its [type representative](#type-representatives).
+The `fromEnum` and `toEnum` functions both take one argument:
+
+    E.fromEnum(a)
+    E.toEnum(n)
+
+Given a value `a`, one can access its type representative via the
+`constructor` property:
+
+    a.constructor.fromEnum(b)
+    a.constructor.toEnum(n)
+
+1. `b` must be of the same Enum
+
+    1. If `b` is not the same Enum, an exception must be thrown.
```

Comment:
I think I copied this from the Haskell report. Should probably stay consistent with the other algebras